src/main/kotlin/net/weavemc/loader/WeaveLoader.kt

```diff
@@ -46,23 +55,40 @@ public object WeaveLoader {
 
                 val configEntry = jar.getEntry("weave.mod.json") ?: error("${jar.name} does not contain a weave.mod.json!")
                 val config = json.decodeFromStream<ModConfig>(jar.getInputStream(configEntry))
+                val name = config.name ?: jar.name
 
                 config.mixinConfigs.forEach(Mixins::addConfiguration)
                 HookManager.hooks += config.hooks.map(::instantiate)
-                initializers += config.entrypoints.map(::instantiate)
+
+                // TODO: Add a name field to the config.
+                mods[config.modId] = WeaveMod(config.entrypoints.map(::instantiate), name, config)
             }
 
-        // call preInit after all hooks/mixins are added
-        initializers.forEach { it.preInit() }
+        // Call preInit() once everything is done.
+        mods.values.forEach {
+            it.instance.forEach(ModInitializer::preInit)
+        }
 
         println("[Weave] Initialized Weave")
     }
 
+    /**
```

Comment:
Documentation should use inline syntax for KDoc comments: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments